src/x/dockertest/docker_resource.go

```diff
@@ -65,11 +65,33 @@ func NewDockerResource(
 		)
 	)
 
-	opts := exposePorts(newOptions(containerName), portList)
+	// TODO: this seems hard to use; a different method might be more appropriate.
+	if image.Name == "" {
```

Comment:
Yeah this logic is quite "hidden" from the function's arguments, but I appreciate that this is the existing behavior... Thanks for adding the function comment.